Userland/Libraries/LibGfx/BMPWriter.cpp

```diff
@@ -72,7 +72,7 @@ ErrorOr<ByteBuffer> BMPWriter::encode(Bitmap const& bitmap, Options options)
     return BMPWriter().dump(bitmap, options);
 }
 
-ByteBuffer BMPWriter::compress_pixel_data(ByteBuffer const& pixel_data, BMPWriter::Compression compression)
+ByteBuffer BMPWriter::compress_pixel_data(ByteBuffer pixel_data, BMPWriter::Compression compression)
```

Comment:
I don't think that's a good change - currently this function is a lie and doesn't compress anything, but once it does, it won't need an owned buffer to return directly. I'd rather keep this as a read-only view (should use `ReadonlyBytes`, really) and copy only in *some* cases (eventually).

Reply:
If you want to avoid the pointless copy right now, we could also get rid of the function entirely and leave a FIXME at the current call site.

Reply:
Once it does, it'll still do the same for these compression methods. For others, it'll return a new buffer. But for these types, it won't change, I think?

(Perf impact of this change:

```
% ministat before.txt after.txt                                                                               
x before.txt
+ after.txt
    N           Min           Max        Median           Avg        Stddev
x  10    0.34247899    0.36247087    0.34811425    0.34858809  0.0054117636
+  10    0.33830118    0.34571385    0.34141994    0.34102957   0.002047055
Difference at 95.0% confidence
	-0.00755851 +/- 0.00384418
	-2.16832% +/- 1.10278%
	(Student's t, pooled s = 0.00409131)
```
)

Reply:
I think ideally we'd only call that function for compression methods that actually compress, but I don't feel strongly enough about it to make you change that :)